### ISSUE TITLE
ts-node cannot transpile on nodeJS 20

### ISSUE THREAD

**Issue by wataruian** [oclif/oclif#1246]
Do you want to request a *feature* or report a *bug*?
-----------------------------------------------------
I'm not sure if this is intended behavior or a bug

What is the current behavior?
-----------------------------
When using ts-node with oclif, warning appears and the existing compiled source is executed instead
`Warning: ts-node executable cannot transpile ESM in Node 20. Existing compiled source will be used instead.`

This affects the development experience because you will need to compile in order to test existing code.

Will it be possible to use `tsx` instead of `ts-node`?

What is the expected behavior?
------------------------------
Able to transpile code in dev mode.

"@oclif/core": "^3.15.1"
Node: v20.10.0

**Comment by mdonnalley** [comment]
@wataruian Yep, this a warning we intentionally emit [here](https://github.com/oclif/core/blob/3.18.1/src/config/ts-node.ts#L224)

The TL;DR of it is that Node changed some things in v20 that broke how ts-node's esm loader was working. Because of that, we're not able to compile ESM at runtime when using node 20.

**However** it is possible to use tsx. You just need to replace the shebang in your `bin/dev.js` from ts-node to tsx
and everything should still work as expected. If you find any odd behavior using tsx - feel free to post a new issue or in this [PR](https://github.com/oclif/core/pull/890) that's been hanging around for a while.

Here's a reading list in case you're curious:
* https://github.com/oclif/core/issues/817
* https://nodejs.org/en/blog/announcements/v20-release-announce#custom-esm-loader-hooks-nearing-stable
* https://github.com/TypeStrong/ts-node/issues/1997

If it's okay with you, I'm going to close this issue since we already have an [issue](https://github.com/oclif/core/issues/817) about ts-node + ESM + node 20 and there's a viable workaround - but please reopen it if you feel otherwise. Thanks!

**Comment by mdonnalley** [comment]
Just realized that the oclif/core issue about this was closed. So I'm going to keep this one open and pin it since others are likely to run into this

**Comment by mdonnalley** [comment]
Closing this since users can switch to tsx or bun (https://github.com/oclif/core/pull/966) or use a different node version
